pkg/gpu/stream.go

```diff
@@ -131,6 +144,20 @@ type enrichedKernelLaunch struct {
 
 var errFatbinParsingDisabled = errors.New("fatbin parsing is disabled")
 
+var enrichedKernelLaunchPool ddsync.Pool[enrichedKernelLaunch]
+var kernelSpanPool ddsync.Pool[kernelSpan]
+var memorySpanPool ddsync.Pool[memorySpan]
```

Comment:
why do we have them as global vars and not part of some struct? feels like we are missing some layer that will hold those and the sync.Once declared below